api/client/service/update.go

```diff
@@ -388,6 +388,20 @@ func portConfigToString(portConfig *swarm.PortConfig) string {
 	return fmt.Sprintf("%v/%s", portConfig.PublishedPort, protocol)
 }
 
+func validatePort(val string) (string, error) {
+	proto, port := nat.SplitProtoPort(val)
+	if proto != "tcp" && proto != "udp" {
+		return "", fmt.Errorf("invalid protocol '%s' for %s", proto, val)
+	}
+	if strings.Contains(port, ":") {
```

Comment:
Why wouldn't `nat.ParsePort` pick this up?
Also seems a little weird to check for only one thing instead of using a regex to match a specific pattern(s)